keps/672-serial-job-execution/README.md

```diff
@@ -0,0 +1,448 @@
+# KEP-672: Serial Job Execution with StartupPolicy API
+
+<!--
+This is the title of your KEP. Keep it short, simple, and descriptive. A good
+title can help communicate what the KEP is and should be considered as part of
+any review.
+-->
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a KEP and for
+highlighting any additional information provided beyond the standard KEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API Details](#api-details)
+  - [Implementation](#implementation)
+  - [Quota Management](#quota-management)
+  - [Defaulting/Validation](#defaultingvalidation)
+  - [User Experience](#user-experience)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+  - [Manage quota for Job sequence](#manage-quota-for-job-sequence)
+  - [Support complex DAGs with JobSet](#support-complex-dags-with-jobset)
+- [Alternatives](#alternatives)
+  - [Using workflow engine to execute sequence of jobs](#using-workflow-engine-to-execute-sequence-of-jobs)
+  <!-- /toc -->
+
+## Summary
+
+This KEP outlines the proposal to expand the scope of the StartupPolicy API for JobSet. The
+StartupPolicy API should support to run sequence of ReplicatedJobs after they reach Ready or
+Succeeded status.
+
+## Motivation
+
+Currently, JobSet supports the StartupPolicy API which allows to create Jobs in order after the
+first Job is in ready status. This can be useful when driver should be ready before workers.
+However, sometime high performance computing and machine learning users want to run sequence of
+ReplicatedJobs within JobSet. For example, it is common to run pre-processing,
+distributed fine-tuning, post-processing for LLM fine-tuning.
+
+### Goals
+
+- Add support for serial Job execution for the StartupPolicy API.
+
+### Non-Goals
+
+- Support workflow management like DAGs outside of Job sequence.
+  - Users should consider to use Argo Workflows or Tekton Pipelines for such use-cases.
+- Allowing for a percentage of Jobs in a ReplicatedJob to be ready to consider the
+  whole ReplicatedJob to ready.
+
```

Comment:
Let's say that we do not aim to introduce any other status other than "Succeeded" and "Ready".